### ISSUE TITLE
viewrendered AttributeError: 'QDockWidget' object has no attribute 'indexOf'

### ISSUE THREAD

**Issue by LewisNeal** [leo-editor/leo-editor#1204]
Running commit c3455bb, I have a leo file where the Render pane is not visible.
Running vr-expand command gives this log output:
_Traceback (most recent call last):

  File "N:\git\leo-editor\leo\core\leoCommands.py", line 2077, in executeAnyCommand
return command(event)

  File "N:\git\leo-editor\leo\plugins\viewrendered.py", line 431, in expand_rendering_pane
vr.expand()

  File "N:\git\leo-editor\leo\plugins\viewrendered.py", line 722, in expand
self.change_size(100)

  File "N:\git\leo-editor\leo\plugins\viewrendered.py", line 727, in change_size
i = splitter.indexOf(self)

AttributeError: 'QDockWidget' object has no attribute 'indexOf'
[end]_

Here are log pane responses after giving other viewrendered commands:
vr-lock   _rendering pane locked_
vr-unlock  _rendering pane unlocked_
vr-show   (focus switches to body, but no other change)

Leo Log Window
Leo 6.0-devel, devel branch
Python 3.7.3, PyQt version 5.12.3
Windows 10 AMD64 (build 10.0.17134) SP0

**Comment by edreamleo** [comment]
Fixed at 2ddebb1 in devel.

**Comment by edreamleo** [comment]
@LewisNeal Thanks for your continued testing. You uncovered a nest of bugs with this report.

**Comment by LewisNeal** [comment]
That's good news, and all without a screenshot :)
Running commit 2ddebb177 the log Tracebacks are gone however I still cannot get the Render pane to display with this leo file. I only observed this problem with c3455bb. I have not had this problem with any other leo file.

I noticed that if I right-click over the title bar of the Body pane, or title bar of any pane, there is a menu of items:
 Render
>Tabs
>Body
>Minibuffer
>IconBar

All of these are ticked except for Render. Clicking the Render line does nothing.
Only Minibuffer and IconBar are active, and they toggle on/off correctly as expected.
There is no response if I select Tabs or Body.

**Comment by edreamleo** [comment]
On Sat, Jun 15, 2019 at 8:37 AM LewisNeal <notifications@github.com> wrote:

> That's good news, and all without a screenshot :)
>
Hehe.

> I noticed that if I right-click over the title bar of the Body pane, or
> title bar of any pane, there is a menu of items:
>
You've said the secret word!  (You have to be really old to understand the
reference.)

Rev fed7f7b in devel makes *all* docks "closeable", except the central
widget.  Previously, I didn't know how to reopen them, but now I do: just
right-click on the title bar of any remaining dock. Select the desired
dock, and voila!

Btw, this shows that we don't want to remove title bars in docks :-)

*Note*: Leo will *not* automatically show a hidden dock just because it is
in use.  So hiding the Tabs dock or the minibuffer is not going to be a
common thing to do ;-)

Finally, Qt will automagically remembers whether docks were hidden when you
last saved Leo, and will hide the hidden docks when you restart Leo.

In short, #1167 <https://github.com/leo-editor/leo-editor/issues/1167> is
complete.  Sweet.

Edward
